server/controller/index.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+const fse = require('fs-extra');
+const path = require('path');
+
+// Retrieve all files in directory
+const getAllFiles = (dir, allFiles) => {
+  allFiles = allFiles || [];
+  const dirFiles = fse.readdirSync(dir);
+
+  dirFiles.forEach((dirFile) => {
+    const currentDir = path.join(dir, dirFile);
+    fse.statSync(currentDir).isDirectory(dir)
+      ? allFiles = getAllFiles(currentDir, allFiles)
+      : allFiles.push(currentDir);
+  });
+
+  return allFiles;
+};
+
+const shuffleArray = (arr) => {
+  let length = arr.length;
+
+  while (--length > 0) {
+    const rand = Math.floor(Math.random() * (length + 1));
+    const temp = arr[rand];
+    arr[rand] = arr[length];
+    arr[length] = temp;
+  }
+  return arr;
+};
+
+const avatarController = () => {
+  const allFiles = getAllFiles(path.join(process.cwd(), '/client/src/assets/avatars'))
```

Comment:
The path /client/src means that the server and the client has to be on the same instance right. We may want to scale the frontend and backend independently in the future. For e.g the React app could be hosted on an S3 bucket or Netlify.

How about just sending the avatar ids from the server to client and the client uses local images to map it to the actual image?

Reply:
Yes :+1: